django/db/models/lookups.py

```diff
@@ -12,6 +12,7 @@
 class Lookup:
     lookup_name = None
     prepare_rhs = True
+    can_use_none_as_rhs = False
```

Comment:
This attribute will need documenting (in the Lookup API Reference) and it should probably have _some_ test coverage outside of the Postgres specific JSON tests.

Reply:
I'm not sure if that must be a public API -- prepare_rhs isn't documented. I also had the concern that it isn't tested besides via contrib.postgres but I'm not sure as I haven't studied the details much.

Reply:
Ah, OK. Yes. We could go that way.

The inline comment should be enough for someone implementing a similar field: 

https://github.com/Stranger6667/django/blob/d2ad573102864423a17a47196f4a037a8c17f489/django/db/models/sql/query.py#L1086-L1089

(I couldn't immediately think of a non-postgres usage.)

If we're happy with that then my only real query is whether we the two test cases are all we need.